lib/sol/SolRpc.js

```diff
@@ -76,73 +75,82 @@ class SolRPC {
-        const confirmation = await this.connection.confirmTransaction({
-          signature: txid,
-          blockhash: block.hash,
-          lastValidBlockHeight: block.height
-        });
-
-        if (confirmation.value.err) { throw new Error(confirmation.value.err); }
-        return txid;
-      }
-      // legacy
-      let transaction = new Web3.Transaction({
-        blockhash: block.hash,
-        feePayer: fromAccount,
-        lastValidBlockHeight: block.height
-      });
-      const transfer = Web3.SystemProgram.transfer({
-        fromPubkey: fromAccount,
-        toPubkey: address,
-        lamports: BigInt(amount)
-      });
-
-      transaction.add(transfer);
-      if (priority) {
-        transaction = await this.addPriorityFee({ transaction });
+        // versioned tx
+        return await this._sendToAddressType0({ address, amount, fromAccount, fromAccountKeypair, priority, block });
+      } else {
+        // legacy
+        return await this._sendToAddressTypeLegacy({ address, amount, fromAccount, fromAccountKeypair, priority, block });
       }
-      return await Web3.sendAndConfirmTransaction(
-        this.connection,
-        transaction,
-        [fromAccountKeypair]
-      );
     } catch (err) {
       this.emitter.emit('failure', err);
       throw err;
     }
   }
 
+  async _sendToAddressType0({ address, amount, fromAccount, fromAccountKeypair, priority, block }) {
+    const instructions = [
+      Web3.SystemProgram.transfer({
+        fromPubkey: fromAccount,
+        toPubkey: address,
+        lamports: BigInt(amount)
+      })
+    ];
+    const message = new Web3.TransactionMessage({
+      payerKey: fromAccount,
+      recentBlockhash: block.hash,
+      instructions
+    }).compileToV0Message();
+    let transaction = new Web3.VersionedTransaction(message);
+    if (priority) {
+      transaction = await this.addPriorityFee({ transaction });
+    }
+    transaction.sign([fromAccountKeypair]);
+
+    const txid = await this.connection.sendTransaction(transaction, { maxRetries: 5 });
+    const confirmation = await this.connection.confirmTransaction({
+      signature: txid,
+      blockhash: block.hash,
+      lastValidBlockHeight: block.height
+    });
+
+    if (confirmation.value.err) { throw new Error(confirmation.value.err); }
+    return txid;
+  }
+
+  async _sendToAddressTypeLegacy({ address, amount, fromAccount, fromAccountKeypair, priority, block }){
+    let transaction = new Web3.Transaction({
+      blockhash: block.hash,
+      feePayer: fromAccount,
+      lastValidBlockHeight: block.height
+    });
+    const transfer = Web3.SystemProgram.transfer({
+      fromPubkey: fromAccount,
+      toPubkey: address,
+      lamports: BigInt(amount)
+    });
+
+    transaction.add(transfer);
+    if (priority) {
+      transaction = await this.addPriorityFee({ transaction });
+    }
+    return await Web3.sendAndConfirmTransaction(
```

Comment:
This is timing out for me, throwing a `TransactionExpiredBlockheightExceededError`. It appears to be because `lastValidBlockHeight` is a past block height. However, the tx succeeds.

Do we even need to wait for confs? I would expect this to broadcast and return a txid immediately and let the check/listening for confs be done separately.